src/instance/util/leafRefUtil.ts

```diff
@@ -2,6 +2,7 @@ import * as _ from 'lodash';
 
 import { Instance, ListChildInstance, LeafInstance } from '../';
 import Path, { isKeyedSegment, ISegmentKeys } from '../Path';
+import { allow } from './';
```

Comment:
This definitely creates a circular dependency between `index.ts` and this one. I'm surprised this doesn't cause problems...

Reply:
Yeah not sure how it doesn't cause a problem, will fix it anyhow